symengine/dict.cpp

```diff
@@ -51,6 +51,18 @@ inline std::ostream& print_vec_rcp(std::ostream& out, T& d)
     return out;
 }
 
+template<class T>
+inline std::ostream& print_set(std::ostream& out, T& d)
+{
+    out << "{";
+    for (auto p = d.begin(); p != d.end(); p++) {
+        if (p != d.begin()) out << ", ";
+        out << (*p).first << ": " << (*p).second;
+    }
+    out << "}";
+    return out;
+}
+
```

Comment:
No exactly same as `print_map`.
There you can access key and value, here we have to deference to get the set element and then get `first` and `second`.